src/transactions/types.ts

```diff
@@ -267,7 +267,7 @@ export type InputSimulateTransactionOptions = {
  */
 export interface InputGenerateSingleSignerRawTransactionData {
   sender: AccountAddressInput;
-  feePayerAddress?: undefined;
+  hasSponsor?: undefined | false;
```

Comment:
not sure you need the `false`? the intention here is that `hasSponsor` is always undefined right? then I think we shouldn't expect/let them set it to false

Reply:
Hm but if user decided to make it clear, and set the `hasSponsor` to false. I would expect that should also be a viable option.

Reply:
why would we want user to specify it if it doesnt make sense in this context as it is always undefined? even if they specify false.

Reply:
Okay I think I see what you mean. Since the Fee payer transaction data is marked as specifically, `hasSponsor: true`, we dont need to let user to specify the false, and its always undefine for other transaction data. 

Will update.